File: api-transacoes/src/ApiTransacoes/Models/Transacao.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyApi.Models
{
    [Table("transacoes")]
    public class Transacao
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [Column("descricao")]
        [MaxLength(255)]
        public string Descricao { get; set; }

        [Column("valor", TypeName = "decimal(10,2)")]
        public decimal Valor { get; set; }

        [Column("data_transacao")]
        public DateTime DataTransacao { get; set; }
    }
}

File: transacoes-api/src/main/java/com/exemplo/transacoes/entity/Transacao.java
package com.exemplo.transacoes.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "transacoes")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transacao {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    
    @Column(nullable = false)
    private String descricao;
    
    @Column(precision = 10, scale = 2)
    private BigDecimal valor;
    
    @Column(name = "data_transacao")
    private LocalDateTime dataTransacao;
    
    @PrePersist
    protected void onCreate() {
        if (dataTransacao == null) {
            dataTransacao = LocalDateTime.now();
        }
    }
}
